pkg/kev/manifest.go

```diff
@@ -181,39 +188,83 @@ func (m *Manifest) ReconcileConfig(envs ...string) (*Manifest, error) {
 	if err != nil {
 		sg := m.UI.StepGroup()
 		defer sg.Done()
-		renderStepError(m.UI, sg.Add(""), renderStepReconcile, err)
+		renderStepError(m.UI, sg.Add(""), renderStepReconcileDetect, err)
 		return nil, err
 	}
 
 	for _, e := range filteredEnvs {
-		if err := validateExtensions(e.override); err != nil {
+		if err := validateEnvExtensions(e, sourcesOverride); err != nil {
 			sg := m.UI.StepGroup()
-			defer sg.Done()
-			renderStepError(m.UI, sg.Add(""), renderStepReconcile, err)
+			renderStepError(m.UI, sg.Add(""), renderStepReconcileDetect, err)
+			sg.Done()
 			return nil, err
 		}
 
 		log.DebugTitlef("Reconciling environment [%s]", e.Name)
 
 		m.UI.Output(fmt.Sprintf("%s: %s", e.Name, e.File))
 
-		sourcesOverride.diffAndPatch(e.override)
+		if err := sourcesOverride.diffAndPatch(e.override); err != nil {
+			sg := m.UI.StepGroup()
+			renderStepError(m.UI, sg.Add(""), renderStepReconcileApply, err)
+			sg.Done()
+			return nil, err
+		}
 	}
 
 	return m, nil
 }
 
-func validateExtensions(override *composeOverride) error {
-	for _, s := range override.Services {
-		_, err := config.ParseSvcK8sConfigFromMap(s.Extensions)
+func validateEnvExtensions(e *Environment, base *composeOverride) error {
+	for _, s := range e.GetServices() {
+		baseSvc, missingSvcErr := base.getService(s.Name)
+		if missingSvcErr != nil {
+			continue
+		}
+
+		baseSvcK8sCfg, err := config.ParseSvcK8sConfigFromMap(baseSvc.Extensions, config.SkipValidation())
+		if err != nil {
+			return errors.Wrapf(missingSvcErr, "when parsing service %s extensions in base override", baseSvc.Name)
+		}
+
+		envSvcK8sCfg, err := config.ParseSvcK8sConfigFromMap(s.Extensions, config.SkipValidation())
 		if err != nil {
-			return errors.Wrapf(err, "when parsing service %s extensions", s.Name)
+			return errors.Wrapf(missingSvcErr, "when parsing service %s extensions", s.Name)
+		}
+
+		mergedK8sSvcCfg, err := baseSvcK8sCfg.Merge(envSvcK8sCfg)
+		if err != nil {
+			return missingSvcErr
+		}
+
+		if err := mergedK8sSvcCfg.Validate(); err != nil {
+			return err
 		}
 	}
-	for name, vol := range override.Volumes {
-		_, err := config.ParseVolK8sConfigFromMap(vol.Extensions)
+
+	for name, vol := range e.GetVolumes() {
+		baseVol, missingVolError := base.getVolume(name)
+		if missingVolError != nil {
+			continue
+		}
+
+		baseVolK8sCfg, err := config.ParseVolK8sConfigFromMap(baseVol.Extensions)
+		if err != nil {
+			return errors.Wrapf(missingVolError, "when parsing vol %s extensions in base override", name)
```

Comment:
similar comment here regarding "base override". I'm guessing we are talking about volume k8s config (via extension) in base compose file?

Reply:
Yes that's exactly right: volume k8s config (via extension) in base compose file.

Reply:
That's what I figured out. Feels like "override" at the end is superfluous?